src/basicfuns.jl

```diff
@@ -18,7 +18,7 @@ xlogx(x::Real) = x > zero(x) ? x * log(x) : zero(log(x))
 
 Return `x * log(y)` for `y > 0` with correct limit at `x = 0`.
 """
-xlogy(x::T, y::T) where {T<:Real} = x > zero(T) ? x * log(y) : zero(log(x))
+xlogy(x::T, y::T) where {T<:Real} = x != zero(T) ? x * log(y) : zero(log(y))
```

Comment:
BTW, maybe you could apply the same change to `xlogx`. That would be more consistent, and more in line with the docstring, which doesn't say that zero is returned for negative inputs at all.

Reply:
Strictly speaking I'm not sure this is necessary.

I'll do it if you really want to, but for `xlogx`, all changing the condition statement from `x > 0` to `x != 0` will do is make it so when `x` is negative the condition returns `true` and the negative log argument error occurs at `x * log(x)` rather than at `zero(log(x))`.

Reply:
Ah, right, I got confused. But for consistency I'd prefer to also change `xlogx`. Otherwise when reading the code people will wonder why there's a difference with `xlogy`.

Reply:
change made.

Reply:
```suggestion
xlogy(x::T, y::T) where {T<:Real} = !iszero(x) ? x * log(y) : zero(log(y))
```

Or even:
```suggestion
xlogy(x, y) = iszero(x) ?  zero(log(y)) : x * log(y)
```